src/test/ui/borrowck/issue-95079-missing-move-in-nested-closure.stderr

```diff
@@ -0,0 +1,31 @@
+error: captured variable cannot escape `FnMut` closure body
+  --> $DIR/issue-95079-missing-move-in-nested-closure.rs:3:29
+   |
+LL | fn foo1(s: &str) -> impl Iterator<Item = String> + '_ {
+   |         - variable defined here
+LL |     None.into_iter()
+LL |         .flat_map(move |()| s.chars().map(|c| format!("{}{}", c, s)))
+   |                           - -^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |                           | |
+   |                           | returns a reference to a captured variable which escapes the closure body
+   |                           | variable captured here
+   |                           inferred to be a `FnMut` closure
+   |
+   = note: `FnMut` closures only have access to their captured variables while they are executing...
+   = note: ...therefore, they cannot allow references to captured variables to escape
+   = help: consider adding 'move' keyword before the parameter of closure in `Map<Chars<'_>, [closure@$DIR/issue-95079-missing-move-in-nested-closure.rs:3:43: 3:46]>`
```

Comment:
This suggestion still doesn't really make sense, or at least it does not draw my eyes to the inner `|c|` closure.

Reply:
Preferably we would have a structured suggestion here, like:

```
move |c| ...
++++
```

but it might not be possible in this stage of borrowck...

Reply:
Yes, It is not clear enough. I will try to refine this by printing out exactly where to add `move` in the source.

I thought if users were aware of the line and column number as [closure@...3:43: 3:46] pointed out, they would be able to fig out where to add `move` :-)